From db0a5a44922f992ca954732dfd4fe26e2509e633 Mon Sep 17 00:00:00 2001
From: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Date: Sat, 26 Apr 2025 10:21:02 +0200
Subject: [PATCH] kernel: bump 6.6 to 6.6.88

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.88

Manually rebased:
 - bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch[1]
 - generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch[2]
 - generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch[3]
 - generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch[4]

All other patches automatically rebased.

1. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=f249c05416ea0bef24c9dbed0e653d2fad87b127
2. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=95f17738b86fd198924d874a5639bcdc49c7e5b8
3. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=2f9761a94bae33d26e6a81b31b36e7d776d93dc1
4. https://web.git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.6.88&id=a2874f0dff63829d1f540003e2d83adb610ee64a

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
---
 ...he-abused-interrupt-map-on-pcie-node.patch |  2 +-
 ...ce-quirks-for-Freeway-Airmouse-T3-an.patch |  2 +-
 ...7-Support-for-the-Sony-IMX477-sensor.patch |  2 +-
 ...9-Support-for-the-Sony-IMX519-sensor.patch |  2 +-
 ...vicetree-Add-documentation-for-imx37.patch |  2 +-
 ...v7251-Add-fwnode-properties-controls.patch |  6 +++---
 ...ov7251-Make-the-enable-GPIO-optional.patch |  6 +++---
 ...-Reinstate-setting-ov7251_global_ini.patch |  2 +-
 ...-Add-module-param-to-select-ext-trig.patch |  2 +-
 ...a-i2c-Add-IMX708-CMOS-sensor-binding.patch |  2 +-
 ...a-i2c-Replace-IMX708-sensor-binding-.patch |  2 +-
 ...a-imx258-Rename-to-include-vendor-pr.patch |  2 +-
 ...0521-PCI-brcmstb-Add-BCM2712-support.patch |  4 ++--
 ...0545-ASoC-dwc-Support-set_bclk_ratio.patch |  4 ++--
 ...950-0546-ASoC-dwc-Add-DMACR-handling.patch |  6 +++---
 ...0-0547-ASOC-dwc-Improve-DMA-shutdown.patch | 14 ++++++-------
 ...8-ASOC-dwc-Fix-16-bit-audio-handling.patch |  4 ++--
 ...ove-check-in-set_bclk_ratio-handling.patch |  2 +-
 ...-Switch-from-V4L2_CID_GAIN-to-V4L2_C.patch |  4 ++--
 ...dwc-Permit-sample-rates-up-to-384kHz.patch |  2 +-
 ...0-0752-ASoC-dwc-Fix-full-duplex-mode.patch |  6 +++---
 ...ndings-i2c-Add-Rohm-BU64754-bindings.patch |  2 +-
 ...2c-Add-driver-for-OmniVision-OV64A40.patch |  2 +-
 ...fer-bclk_ratio-handling-to-hw_params.patch |  6 +++---
 ...efer-bclk_ratio-handling-to-hw_param.patch | 10 +++++-----
 ...-dwc-Correct-channel-count-reporting.patch |  6 +++---
 ...ible-to-apply-PWM-changes-in-atomic-.patch |  2 +-
 ...-Add-Raspberry-Pi-PiSP-Back-End-uAPI.patch |  2 +-
 ...s-Add-bindings-for-Raspberry-Pi-PiSP.patch |  2 +-
 ...-bindings-Add-RPI-RP2040-GPIO-Bridge.patch |  2 +-
 ...river-for-the-RPI-RP2040-GPIO-bridge.patch |  2 +-
 ...s-choose-FIFO-thresholds-based-on-DM.patch | 12 +++++------
 ...edia-dt-bindings-i2c-Add-Sony-IMX500.patch |  2 +-
 ...2c-Add-driver-for-Sony-IMX500-sensor.patch |  2 +-
 ...ment-ignoring-the-hardware-TX_FAULT-.patch |  8 ++++----
 ...hy-add-driver-for-MediaTek-XFI-T-PHY.patch |  2 +-
 ...A-switch-drivers-to-provide-their-ow.patch |  2 +-
 ...rework-the-RollBall-PHY-waiting-code.patch |  8 ++++----
 ...HY-discovery-for-FS-SFP-10G-T-module.patch |  6 +++---
 ...795-v6.7-16-r8152-use-napi_gro_frags.patch |  4 ++--
 ...-nvmem-layouts-add-U-Boot-env-layout.patch |  2 +-
 ...x-rename-mv88e6xxx_g2_scratch_gpio_s.patch |  2 +-
 ...x-add-Amethyst-specific-SMI-GPIO-fun.patch |  2 +-
 ...et-dsa-mv88e6xxx-Support-LED-control.patch | 10 +++++-----
 ...y-to-free-deferred-skbs-while-waitin.patch |  6 ++++--
 ...-dsa-mv88e6xxx-disable-ATU-violation.patch |  2 +-
 ...-r8152-add-LED-configuration-from-OF.patch |  8 ++++----
 target/linux/generic/kernel-6.6               |  4 ++--
 ...add-uImage.FIT-subimage-block-driver.patch |  2 +-
 ...del-do-not-defer-queue-length-update.patch | 17 ++++++----------
 ..._F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch |  4 ++--
 ...-mediatek-enlarge-DMA-reserve-buffer.patch |  2 +-
 ...-add-driver-for-MediaTek-USXGMII-PCS.patch |  2 +-
 ...introduce-the-Qualcomm-IPQESS-driver.patch |  2 +-
 ...dsa-add-out-of-band-tagging-protocol.patch |  2 +-
 ...-lantiq-gswip-convert-to-YAML-schema.patch |  2 +-
 .../701-staging-add-fsl_ppfe-driver.patch     |  2 +-
 ...-pwm-mediatek-add-support-for-MT7988.patch |  4 ++--
 ...an-entry-for-the-IEI-WT61P803-PUZZLE.patch |  2 +-
 .../810-uvc-add-iPassion-iP2970-support.patch |  2 +-
 ...sfp-re-probe-modules-on-DEV_UP-event.patch |  4 ++--
 ...14-net-phy-sfp-add-support-for-SMBus.patch |  6 +++---
 ...NTAINERS-Update-all-StarFive-entries.patch | 20 +++++++++----------
 ...-getting-platform-data-error-for-Sta.patch |  2 +-
 ...-RX-master-support-for-StarFive-JH71.patch |  4 ++--
 ...arfive-Add-JH7100-audio-reset-driver.patch |  2 +-
 66 files changed, 137 insertions(+), 140 deletions(-)

diff --git a/target/linux/ath79/patches-6.6/810-ath79-ignore-the-abused-interrupt-map-on-pcie-node.patch b/target/linux/ath79/patches-6.6/810-ath79-ignore-the-abused-interrupt-map-on-pcie-node.patch
index 330c0d139b33f8..b53461e37f65d0 100644
--- a/target/linux/ath79/patches-6.6/810-ath79-ignore-the-abused-interrupt-map-on-pcie-node.patch
+++ b/target/linux/ath79/patches-6.6/810-ath79-ignore-the-abused-interrupt-map-on-pcie-node.patch
@@ -22,7 +22,7 @@ Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
 
 --- a/drivers/of/irq.c
 +++ b/drivers/of/irq.c
-@@ -89,6 +89,8 @@ EXPORT_SYMBOL_GPL(of_irq_find_parent);
+@@ -94,6 +94,8 @@ EXPORT_SYMBOL_GPL(of_irq_find_parent);
   * drawing board.
   */
  static const char * const of_irq_imap_abusers[] = {
diff --git a/target/linux/bcm27xx/patches-6.6/950-0169-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch b/target/linux/bcm27xx/patches-6.6/950-0169-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
index 9b6617361fa53c..12518a43987ea5 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0169-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0169-hid-usb-Add-device-quirks-for-Freeway-Airmouse-T3-an.patch
@@ -33,7 +33,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.org>
  #define USB_VENDOR_ID_BELKIN		0x050d
  #define USB_DEVICE_ID_FLIP_KVM		0x3201
  
-@@ -1408,6 +1411,9 @@
+@@ -1439,6 +1442,9 @@
  #define USB_VENDOR_ID_XIAOMI		0x2717
  #define USB_DEVICE_ID_MI_SILENT_MOUSE	0x5014
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0173-media-i2c-imx477-Support-for-the-Sony-IMX477-sensor.patch b/target/linux/bcm27xx/patches-6.6/950-0173-media-i2c-imx477-Support-for-the-Sony-IMX477-sensor.patch
index 264b3ca376ac59..08ba07231871e8 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0173-media-i2c-imx477-Support-for-the-Sony-IMX477-sensor.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0173-media-i2c-imx477-Support-for-the-Sony-IMX477-sensor.patch
@@ -343,7 +343,7 @@ Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20058,6 +20058,14 @@ T:	git git://linuxtv.org/media_tree.git
+@@ -20059,6 +20059,14 @@ T:	git git://linuxtv.org/media_tree.git
  F:	Documentation/devicetree/bindings/media/i2c/sony,imx415.yaml
  F:	drivers/media/i2c/imx415.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0174-media-i2c-imx519-Support-for-the-Sony-IMX519-sensor.patch b/target/linux/bcm27xx/patches-6.6/950-0174-media-i2c-imx519-Support-for-the-Sony-IMX519-sensor.patch
index 2ba00932c27735..240c63509ceb63 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0174-media-i2c-imx519-Support-for-the-Sony-IMX519-sensor.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0174-media-i2c-imx519-Support-for-the-Sony-IMX519-sensor.patch
@@ -177,7 +177,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20066,6 +20066,14 @@ T:	git git://linuxtv.org/media_tree.git
+@@ -20067,6 +20067,14 @@ T:	git git://linuxtv.org/media_tree.git
  F:	Documentation/devicetree/bindings/media/i2c/imx477.yaml
  F:	drivers/media/i2c/imx477.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0175-Documentation-devicetree-Add-documentation-for-imx37.patch b/target/linux/bcm27xx/patches-6.6/950-0175-Documentation-devicetree-Add-documentation-for-imx37.patch
index 019ca418aebb1d..3af5c0e95cb1e9 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0175-Documentation-devicetree-Add-documentation-for-imx37.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0175-Documentation-devicetree-Add-documentation-for-imx37.patch
@@ -132,7 +132,7 @@ Signed-off-by: David Plowman <david.plowman@raspberrypi.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20063,6 +20063,7 @@ M:	Raspberry Pi Kernel Maintenance <kern
+@@ -20064,6 +20064,7 @@ M:	Raspberry Pi Kernel Maintenance <kern
  L:	linux-media@vger.kernel.org
  S:	Maintained
  T:	git git://linuxtv.org/media_tree.git
diff --git a/target/linux/bcm27xx/patches-6.6/950-0296-media-i2c-ov7251-Add-fwnode-properties-controls.patch b/target/linux/bcm27xx/patches-6.6/950-0296-media-i2c-ov7251-Add-fwnode-properties-controls.patch
index 06d19e63fb7765..b8adfd8a77cb23 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0296-media-i2c-ov7251-Add-fwnode-properties-controls.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0296-media-i2c-ov7251-Add-fwnode-properties-controls.patch
@@ -13,7 +13,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/media/i2c/ov7251.c
 +++ b/drivers/media/i2c/ov7251.c
-@@ -1541,7 +1541,7 @@ static int ov7251_init_ctrls(struct ov72
+@@ -1543,7 +1543,7 @@ static int ov7251_init_ctrls(struct ov72
  	s64 pixel_rate;
  	int hblank;
  
@@ -22,7 +22,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  	ov7251->ctrls.lock = &ov7251->lock;
  
  	v4l2_ctrl_new_std(&ov7251->ctrls, &ov7251_ctrl_ops,
-@@ -1600,6 +1600,7 @@ static int ov7251_init_ctrls(struct ov72
+@@ -1602,6 +1602,7 @@ static int ov7251_init_ctrls(struct ov72
  
  static int ov7251_probe(struct i2c_client *client)
  {
@@ -30,7 +30,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  	struct device *dev = &client->dev;
  	struct ov7251 *ov7251;
  	unsigned int rate = 0, clk_rate = 0;
-@@ -1690,6 +1691,15 @@ static int ov7251_probe(struct i2c_clien
+@@ -1692,6 +1693,15 @@ static int ov7251_probe(struct i2c_clien
  		goto destroy_mutex;
  	}
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch b/target/linux/bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch
index c00fc4e09a45cf..c6d6380c1d099c 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0327-media-i2c-ov7251-Make-the-enable-GPIO-optional.patch
@@ -14,13 +14,13 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/media/i2c/ov7251.c
 +++ b/drivers/media/i2c/ov7251.c
-@@ -1676,7 +1676,8 @@ static int ov7251_probe(struct i2c_clien
+@@ -1678,7 +1678,8 @@ static int ov7251_probe(struct i2c_clien
  		return PTR_ERR(ov7251->analog_regulator);
  	}
  
--	ov7251->enable_gpio = devm_gpiod_get(dev, "enable", GPIOD_OUT_HIGH);
+-	ov7251->enable_gpio = devm_gpiod_get(dev, "enable", GPIOD_OUT_LOW);
 +	ov7251->enable_gpio = devm_gpiod_get_optional(dev, "enable",
-+						      GPIOD_OUT_HIGH);
++						      GPIOD_OUT_LOW;
  	if (IS_ERR(ov7251->enable_gpio)) {
  		dev_err(dev, "cannot get enable gpio\n");
  		return PTR_ERR(ov7251->enable_gpio);
diff --git a/target/linux/bcm27xx/patches-6.6/950-0350-media-i2c-ov7251-Reinstate-setting-ov7251_global_ini.patch b/target/linux/bcm27xx/patches-6.6/950-0350-media-i2c-ov7251-Reinstate-setting-ov7251_global_ini.patch
index 50c53c6a719ad1..ca9c0d2c26e44c 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0350-media-i2c-ov7251-Reinstate-setting-ov7251_global_ini.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0350-media-i2c-ov7251-Reinstate-setting-ov7251_global_ini.patch
@@ -15,7 +15,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/media/i2c/ov7251.c
 +++ b/drivers/media/i2c/ov7251.c
-@@ -1344,6 +1344,14 @@ static int ov7251_s_stream(struct v4l2_s
+@@ -1346,6 +1346,14 @@ static int ov7251_s_stream(struct v4l2_s
  		if (ret < 0)
  			goto err_power_down;
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0396-media-i2c-ov7251-Add-module-param-to-select-ext-trig.patch b/target/linux/bcm27xx/patches-6.6/950-0396-media-i2c-ov7251-Add-module-param-to-select-ext-trig.patch
index 09af88873c51d6..7bcd8e41ca789d 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0396-media-i2c-ov7251-Add-module-param-to-select-ext-trig.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0396-media-i2c-ov7251-Add-module-param-to-select-ext-trig.patch
@@ -62,7 +62,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  static const unsigned long supported_xclk_rates[] = {
  	[OV7251_19_2_MHZ] = 19200000,
  	[OV7251_24_MHZ] = 24000000,
-@@ -1372,6 +1384,23 @@ static int ov7251_s_stream(struct v4l2_s
+@@ -1374,6 +1386,23 @@ static int ov7251_s_stream(struct v4l2_s
  			dev_err(ov7251->dev, "could not sync v4l2 controls\n");
  			goto err_power_down;
  		}
diff --git a/target/linux/bcm27xx/patches-6.6/950-0407-dtbindings-media-i2c-Add-IMX708-CMOS-sensor-binding.patch b/target/linux/bcm27xx/patches-6.6/950-0407-dtbindings-media-i2c-Add-IMX708-CMOS-sensor-binding.patch
index 5962bb3fbb9046..986922d7995334 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0407-dtbindings-media-i2c-Add-IMX708-CMOS-sensor-binding.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0407-dtbindings-media-i2c-Add-IMX708-CMOS-sensor-binding.patch
@@ -138,7 +138,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20107,6 +20107,14 @@ T:	git git://linuxtv.org/media_tree.git
+@@ -20108,6 +20108,14 @@ T:	git git://linuxtv.org/media_tree.git
  F:	Documentation/devicetree/bindings/media/i2c/imx519.yaml
  F:	drivers/media/i2c/imx519.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0434-dt-bindings-media-i2c-Replace-IMX708-sensor-binding-.patch b/target/linux/bcm27xx/patches-6.6/950-0434-dt-bindings-media-i2c-Replace-IMX708-sensor-binding-.patch
index 5c8457c8690b4a..f3d68c45461628 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0434-dt-bindings-media-i2c-Replace-IMX708-sensor-binding-.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0434-dt-bindings-media-i2c-Replace-IMX708-sensor-binding-.patch
@@ -271,7 +271,7 @@ Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20112,7 +20112,7 @@ M:	Raspberry Pi Kernel Maintenance <kern
+@@ -20113,7 +20113,7 @@ M:	Raspberry Pi Kernel Maintenance <kern
  L:	linux-media@vger.kernel.org
  S:	Maintained
  T:	git git://linuxtv.org/media_tree.git
diff --git a/target/linux/bcm27xx/patches-6.6/950-0460-dt-bindings-media-imx258-Rename-to-include-vendor-pr.patch b/target/linux/bcm27xx/patches-6.6/950-0460-dt-bindings-media-imx258-Rename-to-include-vendor-pr.patch
index 9a10fe73151c7e..50ae214755b1cf 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0460-dt-bindings-media-imx258-Rename-to-include-vendor-pr.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0460-dt-bindings-media-imx258-Rename-to-include-vendor-pr.patch
@@ -291,7 +291,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 +    };
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20013,7 +20013,7 @@ M:	Sakari Ailus <sakari.ailus@linux.inte
+@@ -20014,7 +20014,7 @@ M:	Sakari Ailus <sakari.ailus@linux.inte
  L:	linux-media@vger.kernel.org
  S:	Maintained
  T:	git git://linuxtv.org/media_tree.git
diff --git a/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch b/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
index 38f2e8631729e9..2d7d457887b5be 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0521-PCI-brcmstb-Add-BCM2712-support.patch
@@ -1019,7 +1019,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		clk_disable_unprepare(pcie->clk);
  		return ret;
  	}
-@@ -1599,6 +1970,33 @@ static int brcm_pcie_probe(struct platfo
+@@ -1604,6 +1975,33 @@ static int brcm_pcie_probe(struct platfo
  			dev_err(pcie->dev, "probe of internal MSI failed");
  			goto fail;
  		}
@@ -1053,7 +1053,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	}
  
  	bridge->ops = pcie->type == BCM7425 ? &brcm7425_pcie_ops : &brcm_pcie_ops;
-@@ -1615,6 +2013,8 @@ static int brcm_pcie_probe(struct platfo
+@@ -1620,6 +2018,8 @@ static int brcm_pcie_probe(struct platfo
  		return ret;
  	}
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0545-ASoC-dwc-Support-set_bclk_ratio.patch b/target/linux/bcm27xx/patches-6.6/950-0545-ASoC-dwc-Support-set_bclk_ratio.patch
index 29077e0d1f6439..bc7cd7ddbaf430 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0545-ASoC-dwc-Support-set_bclk_ratio.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0545-ASoC-dwc-Support-set_bclk_ratio.patch
@@ -10,7 +10,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -460,6 +460,40 @@ static int dw_i2s_set_tdm_slot(struct sn
+@@ -459,6 +459,40 @@ static int dw_i2s_set_tdm_slot(struct sn
  	return 0;
  }
  
@@ -51,7 +51,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  static int dw_i2s_dai_probe(struct snd_soc_dai *dai)
  {
  	struct dw_i2s_dev *dev = snd_soc_dai_get_drvdata(dai);
-@@ -476,6 +510,7 @@ static const struct snd_soc_dai_ops dw_i
+@@ -475,6 +509,7 @@ static const struct snd_soc_dai_ops dw_i
  	.trigger	= dw_i2s_trigger,
  	.set_fmt	= dw_i2s_set_fmt,
  	.set_tdm_slot	= dw_i2s_set_tdm_slot,
diff --git a/target/linux/bcm27xx/patches-6.6/950-0546-ASoC-dwc-Add-DMACR-handling.patch b/target/linux/bcm27xx/patches-6.6/950-0546-ASoC-dwc-Add-DMACR-handling.patch
index 6eba454d5c610f..bc577668ab9111 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0546-ASoC-dwc-Add-DMACR-handling.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0546-ASoC-dwc-Add-DMACR-handling.patch
@@ -14,7 +14,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -248,7 +248,7 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -247,7 +247,7 @@ static void dw_i2s_config(struct dw_i2s_
  {
  	u32 ch_reg;
  	struct i2s_clk_config_data *config = &dev->config;
@@ -23,7 +23,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  
  	i2s_disable_channels(dev, stream);
  
-@@ -260,6 +260,7 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -259,6 +259,7 @@ static void dw_i2s_config(struct dw_i2s_
  				      dev->fifo_th - 1);
  			i2s_write_reg(dev->i2s_base, TER(ch_reg), TER_TXCHEN |
  				      dev->tdm_mask << TER_TXSLOT_SHIFT);
@@ -31,7 +31,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		} else {
  			i2s_write_reg(dev->i2s_base, RCR(ch_reg),
  				      dev->xfer_resolution);
-@@ -267,9 +268,15 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -266,9 +267,15 @@ static void dw_i2s_config(struct dw_i2s_
  				      dev->fifo_th - 1);
  			i2s_write_reg(dev->i2s_base, RER(ch_reg), RER_RXCHEN |
  				      dev->tdm_mask << RER_RXSLOT_SHIFT);
diff --git a/target/linux/bcm27xx/patches-6.6/950-0547-ASOC-dwc-Improve-DMA-shutdown.patch b/target/linux/bcm27xx/patches-6.6/950-0547-ASOC-dwc-Improve-DMA-shutdown.patch
index bbe8c77488d723..f560b694bbd3f7 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0547-ASOC-dwc-Improve-DMA-shutdown.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0547-ASOC-dwc-Improve-DMA-shutdown.patch
@@ -22,7 +22,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  #include <sound/pcm.h>
  #include <sound/pcm_params.h>
  #include <sound/soc.h>
-@@ -208,15 +208,10 @@ static void i2s_start(struct dw_i2s_dev
+@@ -206,15 +206,10 @@ static void i2s_start(struct dw_i2s_dev
  	i2s_write_reg(dev->i2s_base, CER, 1);
  }
  
@@ -37,9 +37,9 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 -	else
 -		i2s_write_reg(dev->i2s_base, IRER, 0);
  
- 	if (dev->use_pio || dev->is_jh7110)
- 		i2s_disable_irqs(dev, substream->stream, 8);
-@@ -225,23 +220,15 @@ static void i2s_stop(struct dw_i2s_dev *
+ 	if (!(dev->use_pio || dev->is_jh7110))
+ 		i2s_disable_dma(dev, substream->stream);
+@@ -224,23 +219,15 @@ static void i2s_stop(struct dw_i2s_dev *
  
  	if (!dev->active) {
  		i2s_write_reg(dev->i2s_base, CER, 0);
@@ -67,7 +67,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  }
  
  static void dw_i2s_config(struct dw_i2s_dev *dev, int stream)
-@@ -354,6 +341,62 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -353,6 +340,62 @@ static int dw_i2s_hw_params(struct snd_p
  	return 0;
  }
  
@@ -130,7 +130,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  static int dw_i2s_prepare(struct snd_pcm_substream *substream,
  			  struct snd_soc_dai *dai)
  {
-@@ -381,9 +424,12 @@ static int dw_i2s_trigger(struct snd_pcm
+@@ -380,9 +423,12 @@ static int dw_i2s_trigger(struct snd_pcm
  		i2s_start(dev, substream);
  		break;
  
@@ -144,7 +144,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		dev->active--;
  		i2s_stop(dev, substream);
  		break;
-@@ -512,6 +558,7 @@ static int dw_i2s_dai_probe(struct snd_s
+@@ -511,6 +557,7 @@ static int dw_i2s_dai_probe(struct snd_s
  static const struct snd_soc_dai_ops dw_i2s_dai_ops = {
  	.probe		= dw_i2s_dai_probe,
  	.startup	= dw_i2s_startup,
diff --git a/target/linux/bcm27xx/patches-6.6/950-0548-ASOC-dwc-Fix-16-bit-audio-handling.patch b/target/linux/bcm27xx/patches-6.6/950-0548-ASOC-dwc-Fix-16-bit-audio-handling.patch
index b8a1849dbe01fc..d5f3adaf5b80dd 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0548-ASOC-dwc-Fix-16-bit-audio-handling.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0548-ASOC-dwc-Fix-16-bit-audio-handling.patch
@@ -22,7 +22,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -271,23 +271,34 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -270,23 +270,34 @@ static int dw_i2s_hw_params(struct snd_p
  {
  	struct dw_i2s_dev *dev = snd_soc_dai_get_drvdata(dai);
  	struct i2s_clk_config_data *config = &dev->config;
@@ -57,7 +57,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		dev->ccr = 0x10;
  		dev->xfer_resolution = 0x05;
  		break;
-@@ -519,24 +530,21 @@ static int dw_i2s_set_bclk_ratio(struct
+@@ -518,24 +529,21 @@ static int dw_i2s_set_bclk_ratio(struct
  	struct dw_i2s_dev *dev = snd_soc_dai_get_drvdata(cpu_dai);
  	struct i2s_clk_config_data *config = &dev->config;
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0732-ASoC-dwc-Remove-check-in-set_bclk_ratio-handling.patch b/target/linux/bcm27xx/patches-6.6/950-0732-ASoC-dwc-Remove-check-in-set_bclk_ratio-handling.patch
index d6eb8cbbaa0a26..0112ecad7879e4 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0732-ASoC-dwc-Remove-check-in-set_bclk_ratio-handling.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0732-ASoC-dwc-Remove-check-in-set_bclk_ratio-handling.patch
@@ -18,7 +18,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -528,11 +528,8 @@ static int dw_i2s_set_bclk_ratio(struct
+@@ -527,11 +527,8 @@ static int dw_i2s_set_bclk_ratio(struct
  				 unsigned int ratio)
  {
  	struct dw_i2s_dev *dev = snd_soc_dai_get_drvdata(cpu_dai);
diff --git a/target/linux/bcm27xx/patches-6.6/950-0737-media-i2c-ov7251-Switch-from-V4L2_CID_GAIN-to-V4L2_C.patch b/target/linux/bcm27xx/patches-6.6/950-0737-media-i2c-ov7251-Switch-from-V4L2_CID_GAIN-to-V4L2_C.patch
index 31e1a5cfd4e203..f71ca4cdf75af0 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0737-media-i2c-ov7251-Switch-from-V4L2_CID_GAIN-to-V4L2_C.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0737-media-i2c-ov7251-Switch-from-V4L2_CID_GAIN-to-V4L2_C.patch
@@ -18,7 +18,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
 
 --- a/drivers/media/i2c/ov7251.c
 +++ b/drivers/media/i2c/ov7251.c
-@@ -1063,7 +1063,7 @@ static int ov7251_s_ctrl(struct v4l2_ctr
+@@ -1065,7 +1065,7 @@ static int ov7251_s_ctrl(struct v4l2_ctr
  	case V4L2_CID_EXPOSURE:
  		ret = ov7251_set_exposure(ov7251, ctrl->val);
  		break;
@@ -27,7 +27,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  		ret = ov7251_set_gain(ov7251, ctrl->val);
  		break;
  	case V4L2_CID_TEST_PATTERN:
-@@ -1588,7 +1588,7 @@ static int ov7251_init_ctrls(struct ov72
+@@ -1590,7 +1590,7 @@ static int ov7251_init_ctrls(struct ov72
  	ov7251->exposure = v4l2_ctrl_new_std(&ov7251->ctrls, &ov7251_ctrl_ops,
  					     V4L2_CID_EXPOSURE, 1, 32, 1, 32);
  	ov7251->gain = v4l2_ctrl_new_std(&ov7251->ctrls, &ov7251_ctrl_ops,
diff --git a/target/linux/bcm27xx/patches-6.6/950-0750-ASoC-dwc-Permit-sample-rates-up-to-384kHz.patch b/target/linux/bcm27xx/patches-6.6/950-0750-ASoC-dwc-Permit-sample-rates-up-to-384kHz.patch
index 4a7aaad3a228fc..8bf58f54cfe0cc 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0750-ASoC-dwc-Permit-sample-rates-up-to-384kHz.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0750-ASoC-dwc-Permit-sample-rates-up-to-384kHz.patch
@@ -15,7 +15,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -796,7 +796,7 @@ static int dw_configure_dai_by_dt(struct
+@@ -795,7 +795,7 @@ static int dw_configure_dai_by_dt(struct
  	u32 idx2;
  	int ret;
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0752-ASoC-dwc-Fix-full-duplex-mode.patch b/target/linux/bcm27xx/patches-6.6/950-0752-ASoC-dwc-Fix-full-duplex-mode.patch
index cffdd3302722e9..d1814f55fa5c69 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0752-ASoC-dwc-Fix-full-duplex-mode.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0752-ASoC-dwc-Fix-full-duplex-mode.patch
@@ -15,7 +15,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -235,10 +235,17 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -234,10 +234,17 @@ static void dw_i2s_config(struct dw_i2s_
  {
  	u32 ch_reg;
  	struct i2s_clk_config_data *config = &dev->config;
@@ -34,7 +34,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	for (ch_reg = 0; ch_reg < (config->chan_nr / 2); ch_reg++) {
  		if (stream == SNDRV_PCM_STREAM_PLAYBACK) {
  			i2s_write_reg(dev->i2s_base, TCR(ch_reg),
-@@ -258,10 +265,6 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -257,10 +264,6 @@ static void dw_i2s_config(struct dw_i2s_
  			dmacr |= (DMACR_DMAEN_RXCH0 << ch_reg);
  		}
  	}
@@ -45,7 +45,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  
  	i2s_write_reg(dev->i2s_base, I2S_DMACR, dmacr);
  }
-@@ -370,10 +373,13 @@ static int dw_i2s_startup(struct snd_pcm
+@@ -369,10 +372,13 @@ static int dw_i2s_startup(struct snd_pcm
  
  	dw_i2s_config(dev, substream->stream);
  	dmacr = i2s_read_reg(dev->i2s_base, I2S_DMACR);
diff --git a/target/linux/bcm27xx/patches-6.6/950-0767-media-dt-bindings-i2c-Add-Rohm-BU64754-bindings.patch b/target/linux/bcm27xx/patches-6.6/950-0767-media-dt-bindings-i2c-Add-Rohm-BU64754-bindings.patch
index 2d3259092717a1..607954b20948d5 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0767-media-dt-bindings-i2c-Add-Rohm-BU64754-bindings.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0767-media-dt-bindings-i2c-Add-Rohm-BU64754-bindings.patch
@@ -67,7 +67,7 @@ Signed-off-by: Jacopo Mondi <jacopo.mondi@ideasonboard.com>
 +...
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -18620,6 +18620,13 @@ S:	Supported
+@@ -18621,6 +18621,13 @@ S:	Supported
  F:	drivers/iio/light/rohm-bu27008.c
  F:	drivers/iio/light/rohm-bu27034.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0768-media-i2c-Add-driver-for-OmniVision-OV64A40.patch b/target/linux/bcm27xx/patches-6.6/950-0768-media-i2c-Add-driver-for-OmniVision-OV64A40.patch
index 477ac8319ff89f..6075a9d5e71826 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0768-media-i2c-Add-driver-for-OmniVision-OV64A40.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0768-media-i2c-Add-driver-for-OmniVision-OV64A40.patch
@@ -16,7 +16,7 @@ Signed-off-by: Jacopo Mondi <jacopo.mondi@ideasonboard.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -15872,6 +15872,14 @@ S:	Maintained
+@@ -15873,6 +15873,14 @@ S:	Maintained
  T:	git git://linuxtv.org/media_tree.git
  F:	drivers/media/i2c/ov5695.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0809-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_params.patch b/target/linux/bcm27xx/patches-6.6/950-0809-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_params.patch
index 49ca7d6337265e..17b1b32263b67f 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0809-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_params.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0809-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_params.patch
@@ -18,7 +18,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -314,6 +314,25 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -313,6 +313,25 @@ static int dw_i2s_hw_params(struct snd_p
  	if (dev->tdm_slots)
  		config->data_width = 32;
  
@@ -44,7 +44,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	config->chan_nr = params_channels(params);
  
  	switch (config->chan_nr) {
-@@ -537,23 +556,7 @@ static int dw_i2s_set_bclk_ratio(struct
+@@ -536,23 +555,7 @@ static int dw_i2s_set_bclk_ratio(struct
  
  	dev_dbg(dev->dev, "%s(%d)\n", __func__, ratio);
  
@@ -69,7 +69,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  
  	return 0;
  }
-@@ -1068,6 +1071,7 @@ static int dw_i2s_probe(struct platform_
+@@ -1067,6 +1070,7 @@ static int dw_i2s_probe(struct platform_
  		}
  	}
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-0931-fixup-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_param.patch b/target/linux/bcm27xx/patches-6.6/950-0931-fixup-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_param.patch
index 9f5c2b3a48dee3..7174cf0a5ae708 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0931-fixup-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_param.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0931-fixup-ASoC-dwc-Defer-bclk_ratio-handling-to-hw_param.patch
@@ -13,7 +13,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -288,21 +288,18 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -287,21 +287,18 @@ static int dw_i2s_hw_params(struct snd_p
  	case SNDRV_PCM_FORMAT_S16_LE:
  		config->data_width = 16;
  		dma_data->dt.addr_width = 2;
@@ -35,7 +35,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		dev->xfer_resolution = 0x05;
  		break;
  
-@@ -314,25 +311,6 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -313,25 +310,6 @@ static int dw_i2s_hw_params(struct snd_p
  	if (dev->tdm_slots)
  		config->data_width = 32;
  
@@ -61,7 +61,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  	config->chan_nr = params_channels(params);
  
  	switch (config->chan_nr) {
-@@ -348,11 +326,31 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -347,11 +325,31 @@ static int dw_i2s_hw_params(struct snd_p
  
  	dw_i2s_config(dev, substream->stream);
  
@@ -95,7 +95,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		if (dev->i2s_clk_cfg) {
  			ret = dev->i2s_clk_cfg(config);
  			if (ret < 0) {
-@@ -360,8 +358,7 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -359,8 +357,7 @@ static int dw_i2s_hw_params(struct snd_p
  				return ret;
  			}
  		} else {
@@ -105,7 +105,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  
  			ret = clk_set_rate(dev->clk, bitclk);
  			if (ret) {
-@@ -370,6 +367,8 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -369,6 +366,8 @@ static int dw_i2s_hw_params(struct snd_p
  				return ret;
  			}
  		}
diff --git a/target/linux/bcm27xx/patches-6.6/950-0951-ASoC-dwc-Correct-channel-count-reporting.patch b/target/linux/bcm27xx/patches-6.6/950-0951-ASoC-dwc-Correct-channel-count-reporting.patch
index 8f54995dc05b06..aaf9e32b4022bb 100644
--- a/target/linux/bcm27xx/patches-6.6/950-0951-ASoC-dwc-Correct-channel-count-reporting.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-0951-ASoC-dwc-Correct-channel-count-reporting.patch
@@ -17,7 +17,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -320,7 +320,7 @@ static int dw_i2s_hw_params(struct snd_p
+@@ -319,7 +319,7 @@ static int dw_i2s_hw_params(struct snd_p
  	case TWO_CHANNEL_SUPPORT:
  		break;
  	default:
@@ -26,7 +26,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		return -EINVAL;
  	}
  
-@@ -708,7 +708,7 @@ static int dw_configure_dai(struct dw_i2
+@@ -707,7 +707,7 @@ static int dw_configure_dai(struct dw_i2
  			idx = 1;
  		dw_i2s_dai->playback.channels_min = MIN_CHANNEL_NUM;
  		dw_i2s_dai->playback.channels_max =
@@ -35,7 +35,7 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
  		dw_i2s_dai->playback.formats = formats[idx];
  		dw_i2s_dai->playback.rates = rates;
  	}
-@@ -722,7 +722,7 @@ static int dw_configure_dai(struct dw_i2
+@@ -721,7 +721,7 @@ static int dw_configure_dai(struct dw_i2
  			idx = 1;
  		dw_i2s_dai->capture.channels_min = MIN_CHANNEL_NUM;
  		dw_i2s_dai->capture.channels_max =
diff --git a/target/linux/bcm27xx/patches-6.6/950-1128-pwm-Make-it-possible-to-apply-PWM-changes-in-atomic-.patch b/target/linux/bcm27xx/patches-6.6/950-1128-pwm-Make-it-possible-to-apply-PWM-changes-in-atomic-.patch
index 2441e8223e7247..3377683128ae0a 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1128-pwm-Make-it-possible-to-apply-PWM-changes-in-atomic-.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1128-pwm-Make-it-possible-to-apply-PWM-changes-in-atomic-.patch
@@ -48,7 +48,7 @@ Signed-off-by: Thierry Reding <thierry.reding@gmail.com>
  drivers cannot. If you rely on getting the inactive state, use .duty_cycle=0,
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -17438,7 +17438,7 @@ F:	drivers/video/backlight/pwm_bl.c
+@@ -17439,7 +17439,7 @@ F:	drivers/video/backlight/pwm_bl.c
  F:	include/dt-bindings/pwm/
  F:	include/linux/pwm.h
  F:	include/linux/pwm_backlight.h
diff --git a/target/linux/bcm27xx/patches-6.6/950-1149-media-uapi-Add-Raspberry-Pi-PiSP-Back-End-uAPI.patch b/target/linux/bcm27xx/patches-6.6/950-1149-media-uapi-Add-Raspberry-Pi-PiSP-Back-End-uAPI.patch
index a95742b475e026..a2e36d50d62310 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1149-media-uapi-Add-Raspberry-Pi-PiSP-Back-End-uAPI.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1149-media-uapi-Add-Raspberry-Pi-PiSP-Back-End-uAPI.patch
@@ -23,7 +23,7 @@ Signed-off-by: Jacopo Mondi <jacopo.mondi@ideasonboard.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -18033,6 +18033,13 @@ L:	linux-wireless@vger.kernel.org
+@@ -18034,6 +18034,13 @@ L:	linux-wireless@vger.kernel.org
  S:	Orphan
  F:	drivers/net/wireless/legacy/ray*
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-1152-media-dt-bindings-Add-bindings-for-Raspberry-Pi-PiSP.patch b/target/linux/bcm27xx/patches-6.6/950-1152-media-dt-bindings-Add-bindings-for-Raspberry-Pi-PiSP.patch
index e6f15fa47151e8..12f91d97845e3e 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1152-media-dt-bindings-Add-bindings-for-Raspberry-Pi-PiSP.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1152-media-dt-bindings-Add-bindings-for-Raspberry-Pi-PiSP.patch
@@ -87,7 +87,7 @@ Reviewed-by: Naushir Patuck <naush@raspberrypi.com>
 +    };
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -18038,6 +18038,7 @@ M:	Jacopo Mondi <jacopo.mondi@ideasonboa
+@@ -18039,6 +18039,7 @@ M:	Jacopo Mondi <jacopo.mondi@ideasonboa
  L:	Raspberry Pi Kernel Maintenance <kernel-list@raspberrypi.com>
  L:	linux-media@vger.kernel.org
  S:	Maintained
diff --git a/target/linux/bcm27xx/patches-6.6/950-1164-spi-dt-bindings-Add-RPI-RP2040-GPIO-Bridge.patch b/target/linux/bcm27xx/patches-6.6/950-1164-spi-dt-bindings-Add-RPI-RP2040-GPIO-Bridge.patch
index abdf8157a5486d..785b36e3cf8082 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1164-spi-dt-bindings-Add-RPI-RP2040-GPIO-Bridge.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1164-spi-dt-bindings-Add-RPI-RP2040-GPIO-Bridge.patch
@@ -94,7 +94,7 @@ Signed-off-by: Richard Oliver <richard.oliver@raspberrypi.com>
 +
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -18028,6 +18028,11 @@ F:	drivers/ras/
+@@ -18029,6 +18029,11 @@ F:	drivers/ras/
  F:	include/linux/ras.h
  F:	include/ras/ras_event.h
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-1165-spi-Add-a-driver-for-the-RPI-RP2040-GPIO-bridge.patch b/target/linux/bcm27xx/patches-6.6/950-1165-spi-Add-a-driver-for-the-RPI-RP2040-GPIO-bridge.patch
index db3b26c882bcee..b5f5014b074d33 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1165-spi-Add-a-driver-for-the-RPI-RP2040-GPIO-bridge.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1165-spi-Add-a-driver-for-the-RPI-RP2040-GPIO-bridge.patch
@@ -25,7 +25,7 @@ Signed-off-by: Richard Oliver <richard.oliver@raspberrypi.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -18032,6 +18032,7 @@ RASPBERRY PI RP2040 GPIO BRIDGE DRIVER
+@@ -18033,6 +18033,7 @@ RASPBERRY PI RP2040 GPIO BRIDGE DRIVER
  M:	Raspberry Pi Kernel Maintenance <kernel-list@raspberrypi.com>
  S:	Maintained
  F:	Documentation/devicetree/bindings/spi/raspberrypi,rp2040-gpio-bridge.yaml
diff --git a/target/linux/bcm27xx/patches-6.6/950-1197-sound-soc-dwc-i2s-choose-FIFO-thresholds-based-on-DM.patch b/target/linux/bcm27xx/patches-6.6/950-1197-sound-soc-dwc-i2s-choose-FIFO-thresholds-based-on-DM.patch
index d0e3bc1fab0f1f..18bd7657013373 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1197-sound-soc-dwc-i2s-choose-FIFO-thresholds-based-on-DM.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1197-sound-soc-dwc-i2s-choose-FIFO-thresholds-based-on-DM.patch
@@ -22,7 +22,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -236,6 +236,8 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -235,6 +235,8 @@ static void dw_i2s_config(struct dw_i2s_
  	u32 ch_reg;
  	struct i2s_clk_config_data *config = &dev->config;
  	u32 dmacr;
@@ -31,7 +31,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  
  	i2s_disable_channels(dev, stream);
  
-@@ -251,7 +253,7 @@ static void dw_i2s_config(struct dw_i2s_
+@@ -250,7 +252,7 @@ static void dw_i2s_config(struct dw_i2s_
  			i2s_write_reg(dev->i2s_base, TCR(ch_reg),
  				      dev->xfer_resolution);
  			i2s_write_reg(dev->i2s_base, TFCR(ch_reg),
@@ -40,7 +40,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  			i2s_write_reg(dev->i2s_base, TER(ch_reg), TER_TXCHEN |
  				      dev->tdm_mask << TER_TXSLOT_SHIFT);
  			dmacr |= (DMACR_DMAEN_TXCH0 << ch_reg);
-@@ -783,8 +785,8 @@ static int dw_configure_dai_by_pd(struct
+@@ -782,8 +784,8 @@ static int dw_configure_dai_by_pd(struct
  		dev->capture_dma_data.pd.data = pdata->capture_dma_data;
  		dev->play_dma_data.pd.addr = res->start + I2S_TXDMA;
  		dev->capture_dma_data.pd.addr = res->start + I2S_RXDMA;
@@ -51,7 +51,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  		dev->play_dma_data.pd.addr_width = bus_widths[idx];
  		dev->capture_dma_data.pd.addr_width = bus_widths[idx];
  		dev->play_dma_data.pd.filter = pdata->filter;
-@@ -815,7 +817,10 @@ static int dw_configure_dai_by_dt(struct
+@@ -814,7 +816,10 @@ static int dw_configure_dai_by_dt(struct
  		dev->play_dma_data.dt.addr = res->start + I2S_TXDMA;
  		dev->play_dma_data.dt.fifo_size = fifo_depth *
  			(fifo_width[idx2]) >> 8;
@@ -63,7 +63,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	}
  	if (COMP1_RX_ENABLED(comp1)) {
  		idx2 = COMP2_RX_WORDSIZE_0(comp2);
-@@ -824,9 +829,14 @@ static int dw_configure_dai_by_dt(struct
+@@ -823,9 +828,14 @@ static int dw_configure_dai_by_dt(struct
  		dev->capture_dma_data.dt.addr = res->start + I2S_RXDMA;
  		dev->capture_dma_data.dt.fifo_size = fifo_depth *
  			(fifo_width[idx2] >> 8);
@@ -79,7 +79,7 @@ Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
  	return 0;
  
  }
-@@ -1070,6 +1080,7 @@ static int dw_i2s_probe(struct platform_
+@@ -1069,6 +1079,7 @@ static int dw_i2s_probe(struct platform_
  		}
  	}
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-1243-media-dt-bindings-i2c-Add-Sony-IMX500.patch b/target/linux/bcm27xx/patches-6.6/950-1243-media-dt-bindings-i2c-Add-Sony-IMX500.patch
index cfd4b46d4ff6b7..d4979f73e018b2 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1243-media-dt-bindings-i2c-Add-Sony-IMX500.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1243-media-dt-bindings-i2c-Add-Sony-IMX500.patch
@@ -150,7 +150,7 @@ Signed-off-by: Richard Oliver <richard.oliver@raspberrypi.com>
 +
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20128,6 +20128,13 @@ F:	Documentation/devicetree/bindings/med
+@@ -20129,6 +20129,13 @@ F:	Documentation/devicetree/bindings/med
  F:	Documentation/devicetree/bindings/media/i2c/imx477.yaml
  F:	drivers/media/i2c/imx477.c
  
diff --git a/target/linux/bcm27xx/patches-6.6/950-1244-media-i2c-Add-driver-for-Sony-IMX500-sensor.patch b/target/linux/bcm27xx/patches-6.6/950-1244-media-i2c-Add-driver-for-Sony-IMX500-sensor.patch
index c6b9196a525267..12e5e2b9e4d227 100644
--- a/target/linux/bcm27xx/patches-6.6/950-1244-media-i2c-Add-driver-for-Sony-IMX500-sensor.patch
+++ b/target/linux/bcm27xx/patches-6.6/950-1244-media-i2c-Add-driver-for-Sony-IMX500-sensor.patch
@@ -21,7 +21,7 @@ Signed-off-by: Richard Oliver <richard.oliver@raspberrypi.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20134,6 +20134,7 @@ L:	linux-media@vger.kernel.org
+@@ -20135,6 +20135,7 @@ L:	linux-media@vger.kernel.org
  S:	Maintained
  T:	git git://linuxtv.org/media_tree.git
  F:	Documentation/devicetree/bindings/media/i2c/sony,imx500.yaml
diff --git a/target/linux/generic/backport-6.6/730-v6.7-net-sfp-re-implement-ignoring-the-hardware-TX_FAULT-.patch b/target/linux/generic/backport-6.6/730-v6.7-net-sfp-re-implement-ignoring-the-hardware-TX_FAULT-.patch
index 9cb491b2d84f8b..e1aa6d1907a477 100644
--- a/target/linux/generic/backport-6.6/730-v6.7-net-sfp-re-implement-ignoring-the-hardware-TX_FAULT-.patch
+++ b/target/linux/generic/backport-6.6/730-v6.7-net-sfp-re-implement-ignoring-the-hardware-TX_FAULT-.patch
@@ -43,7 +43,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  }
  
  // For 10GBASE-T short-reach modules
-@@ -796,7 +796,8 @@ static void sfp_soft_start_poll(struct s
+@@ -798,7 +798,8 @@ static void sfp_soft_start_poll(struct s
  
  	mutex_lock(&sfp->st_mutex);
  	// Poll the soft state for hardware pins we want to ignore
@@ -53,7 +53,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  	if (sfp->state_soft_mask & (SFP_F_LOS | SFP_F_TX_FAULT) &&
  	    !sfp->need_poll)
-@@ -2321,7 +2322,7 @@ static int sfp_sm_mod_probe(struct sfp *
+@@ -2323,7 +2324,7 @@ static int sfp_sm_mod_probe(struct sfp *
  	sfp->module_t_start_up = T_START_UP;
  	sfp->module_t_wait = T_WAIT;
  
@@ -62,7 +62,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  	if (sfp->id.base.extended_cc == SFF8024_ECC_10GBASE_T_SFI ||
  	    sfp->id.base.extended_cc == SFF8024_ECC_10GBASE_T_SR ||
-@@ -2344,6 +2345,8 @@ static int sfp_sm_mod_probe(struct sfp *
+@@ -2346,6 +2347,8 @@ static int sfp_sm_mod_probe(struct sfp *
  
  	if (sfp->quirk && sfp->quirk->fixup)
  		sfp->quirk->fixup(sfp);
@@ -71,7 +71,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	mutex_unlock(&sfp->st_mutex);
  
  	return 0;
-@@ -2844,10 +2847,7 @@ static void sfp_check_state(struct sfp *
+@@ -2846,10 +2849,7 @@ static void sfp_check_state(struct sfp *
  	mutex_lock(&sfp->st_mutex);
  	state = sfp_get_state(sfp);
  	changed = state ^ sfp->state;
diff --git a/target/linux/generic/backport-6.6/756-v6.10-phy-add-driver-for-MediaTek-XFI-T-PHY.patch b/target/linux/generic/backport-6.6/756-v6.10-phy-add-driver-for-MediaTek-XFI-T-PHY.patch
index 6afd72d4d200f2..96b5b5dfbb06d3 100644
--- a/target/linux/generic/backport-6.6/756-v6.10-phy-add-driver-for-MediaTek-XFI-T-PHY.patch
+++ b/target/linux/generic/backport-6.6/756-v6.10-phy-add-driver-for-MediaTek-XFI-T-PHY.patch
@@ -63,7 +63,7 @@ Signed-off-by: Vinod Koul <vkoul@kernel.org>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -13366,6 +13366,7 @@ L:	netdev@vger.kernel.org
+@@ -13367,6 +13367,7 @@ L:	netdev@vger.kernel.org
  S:	Maintained
  F:	drivers/net/phy/mediatek-ge-soc.c
  F:	drivers/net/phy/mediatek-ge.c
diff --git a/target/linux/generic/backport-6.6/764-v6.10-net-dsa-allow-DSA-switch-drivers-to-provide-their-ow.patch b/target/linux/generic/backport-6.6/764-v6.10-net-dsa-allow-DSA-switch-drivers-to-provide-their-ow.patch
index 7c9ab16d4af989..b86935f9af7b45 100644
--- a/target/linux/generic/backport-6.6/764-v6.10-net-dsa-allow-DSA-switch-drivers-to-provide-their-ow.patch
+++ b/target/linux/generic/backport-6.6/764-v6.10-net-dsa-allow-DSA-switch-drivers-to-provide-their-ow.patch
@@ -38,7 +38,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  	u32			phys_mii_mask;
 --- a/net/dsa/dsa.c
 +++ b/net/dsa/dsa.c
-@@ -1510,6 +1510,17 @@ static int dsa_switch_probe(struct dsa_s
+@@ -1549,6 +1549,17 @@ static int dsa_switch_probe(struct dsa_s
  	if (!ds->num_ports)
  		return -EINVAL;
  
diff --git a/target/linux/generic/backport-6.6/785-01-v6.8-net-sfp-rework-the-RollBall-PHY-waiting-code.patch b/target/linux/generic/backport-6.6/785-01-v6.8-net-sfp-rework-the-RollBall-PHY-waiting-code.patch
index f113014fa5669c..41d19cb6671671 100644
--- a/target/linux/generic/backport-6.6/785-01-v6.8-net-sfp-rework-the-RollBall-PHY-waiting-code.patch
+++ b/target/linux/generic/backport-6.6/785-01-v6.8-net-sfp-rework-the-RollBall-PHY-waiting-code.patch
@@ -86,7 +86,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  static void sfp_fixup_rollball_cc(struct sfp *sfp)
  {
  	sfp_fixup_rollball(sfp);
-@@ -2320,7 +2318,7 @@ static int sfp_sm_mod_probe(struct sfp *
+@@ -2322,7 +2320,7 @@ static int sfp_sm_mod_probe(struct sfp *
  		mask |= SFP_F_RS1;
  
  	sfp->module_t_start_up = T_START_UP;
@@ -95,7 +95,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  
  	sfp->state_ignore_mask = 0;
  
-@@ -2556,10 +2554,9 @@ static void sfp_sm_main(struct sfp *sfp,
+@@ -2558,10 +2556,9 @@ static void sfp_sm_main(struct sfp *sfp,
  
  		/* We need to check the TX_FAULT state, which is not defined
  		 * while TX_DISABLE is asserted. The earliest we want to do
@@ -108,7 +108,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		break;
  
  	case SFP_S_WAIT:
-@@ -2573,8 +2570,8 @@ static void sfp_sm_main(struct sfp *sfp,
+@@ -2575,8 +2572,8 @@ static void sfp_sm_main(struct sfp *sfp,
  			 * deasserting.
  			 */
  			timeout = sfp->module_t_start_up;
@@ -119,7 +119,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  			else
  				timeout = 1;
  
-@@ -2617,7 +2614,11 @@ static void sfp_sm_main(struct sfp *sfp,
+@@ -2619,7 +2616,11 @@ static void sfp_sm_main(struct sfp *sfp,
  		ret = sfp_sm_probe_for_phy(sfp);
  		if (ret == -ENODEV) {
  			if (--sfp->sm_phy_retries) {
diff --git a/target/linux/generic/backport-6.6/785-02-v6.8-net-sfp-fix-PHY-discovery-for-FS-SFP-10G-T-module.patch b/target/linux/generic/backport-6.6/785-02-v6.8-net-sfp-fix-PHY-discovery-for-FS-SFP-10G-T-module.patch
index 5ba8d0fabc46a3..85a00285c62b2f 100644
--- a/target/linux/generic/backport-6.6/785-02-v6.8-net-sfp-fix-PHY-discovery-for-FS-SFP-10G-T-module.patch
+++ b/target/linux/generic/backport-6.6/785-02-v6.8-net-sfp-fix-PHY-discovery-for-FS-SFP-10G-T-module.patch
@@ -51,7 +51,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  }
  
  static void sfp_fixup_halny_gsfp(struct sfp *sfp)
-@@ -2318,6 +2325,7 @@ static int sfp_sm_mod_probe(struct sfp *
+@@ -2320,6 +2327,7 @@ static int sfp_sm_mod_probe(struct sfp *
  		mask |= SFP_F_RS1;
  
  	sfp->module_t_start_up = T_START_UP;
@@ -59,7 +59,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  	sfp->phy_t_retry = T_PHY_RETRY;
  
  	sfp->state_ignore_mask = 0;
-@@ -2554,9 +2562,10 @@ static void sfp_sm_main(struct sfp *sfp,
+@@ -2556,9 +2564,10 @@ static void sfp_sm_main(struct sfp *sfp,
  
  		/* We need to check the TX_FAULT state, which is not defined
  		 * while TX_DISABLE is asserted. The earliest we want to do
@@ -72,7 +72,7 @@ Signed-off-by: David S. Miller <davem@davemloft.net>
  		break;
  
  	case SFP_S_WAIT:
-@@ -2570,8 +2579,8 @@ static void sfp_sm_main(struct sfp *sfp,
+@@ -2572,8 +2581,8 @@ static void sfp_sm_main(struct sfp *sfp,
  			 * deasserting.
  			 */
  			timeout = sfp->module_t_start_up;
diff --git a/target/linux/generic/backport-6.6/795-v6.7-16-r8152-use-napi_gro_frags.patch b/target/linux/generic/backport-6.6/795-v6.7-16-r8152-use-napi_gro_frags.patch
index 85b320f15aa5b4..73adadc4351df1 100644
--- a/target/linux/generic/backport-6.6/795-v6.7-16-r8152-use-napi_gro_frags.patch
+++ b/target/linux/generic/backport-6.6/795-v6.7-16-r8152-use-napi_gro_frags.patch
@@ -15,7 +15,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 
 --- a/drivers/net/usb/r8152.c
 +++ b/drivers/net/usb/r8152.c
-@@ -2584,8 +2584,9 @@ static int rx_bottom(struct r8152 *tp, i
+@@ -2585,8 +2585,9 @@ static int rx_bottom(struct r8152 *tp, i
  		while (urb->actual_length > len_used) {
  			struct net_device *netdev = tp->netdev;
  			struct net_device_stats *stats = &netdev->stats;
@@ -26,7 +26,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
  
  			WARN_ON_ONCE(skb_queue_len(&tp->rx_queue) >= 1000);
  
-@@ -2598,45 +2599,77 @@ static int rx_bottom(struct r8152 *tp, i
+@@ -2599,45 +2600,77 @@ static int rx_bottom(struct r8152 *tp, i
  				break;
  
  			pkt_len -= ETH_FCS_LEN;
diff --git a/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch b/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch
index af1b8b4e78fb51..2d71c4b3b43fc6 100644
--- a/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch
+++ b/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch
@@ -41,7 +41,7 @@ Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -21989,6 +21989,7 @@ U-BOOT ENVIRONMENT VARIABLES
+@@ -21990,6 +21990,7 @@ U-BOOT ENVIRONMENT VARIABLES
  M:	Rafa Miecki <rafal@milecki.pl>
  S:	Maintained
  F:	Documentation/devicetree/bindings/nvmem/u-boot,env.yaml
diff --git a/target/linux/generic/backport-6.6/896-01-v6.9-net-dsa-mv88e6xxx-rename-mv88e6xxx_g2_scratch_gpio_s.patch b/target/linux/generic/backport-6.6/896-01-v6.9-net-dsa-mv88e6xxx-rename-mv88e6xxx_g2_scratch_gpio_s.patch
index 2c3e99d8e09341..7e71e49dfa33df 100644
--- a/target/linux/generic/backport-6.6/896-01-v6.9-net-dsa-mv88e6xxx-rename-mv88e6xxx_g2_scratch_gpio_s.patch
+++ b/target/linux/generic/backport-6.6/896-01-v6.9-net-dsa-mv88e6xxx-rename-mv88e6xxx_g2_scratch_gpio_s.patch
@@ -19,7 +19,7 @@ Signed-off-by: Paolo Abeni <pabeni@redhat.com>
 
 --- a/drivers/net/dsa/mv88e6xxx/chip.c
 +++ b/drivers/net/dsa/mv88e6xxx/chip.c
-@@ -3699,7 +3699,7 @@ static int mv88e6xxx_mdio_register(struc
+@@ -3725,7 +3725,7 @@ static int mv88e6xxx_mdio_register(struc
  
  	if (external) {
  		mv88e6xxx_reg_lock(chip);
diff --git a/target/linux/generic/backport-6.6/896-02-v6.9-net-dsa-mv88e6xxx-add-Amethyst-specific-SMI-GPIO-fun.patch b/target/linux/generic/backport-6.6/896-02-v6.9-net-dsa-mv88e6xxx-add-Amethyst-specific-SMI-GPIO-fun.patch
index 875acd3ba1671b..4fac0e2a13ae55 100644
--- a/target/linux/generic/backport-6.6/896-02-v6.9-net-dsa-mv88e6xxx-add-Amethyst-specific-SMI-GPIO-fun.patch
+++ b/target/linux/generic/backport-6.6/896-02-v6.9-net-dsa-mv88e6xxx-add-Amethyst-specific-SMI-GPIO-fun.patch
@@ -27,7 +27,7 @@ Signed-off-by: Paolo Abeni <pabeni@redhat.com>
 
 --- a/drivers/net/dsa/mv88e6xxx/chip.c
 +++ b/drivers/net/dsa/mv88e6xxx/chip.c
-@@ -3699,7 +3699,10 @@ static int mv88e6xxx_mdio_register(struc
+@@ -3725,7 +3725,10 @@ static int mv88e6xxx_mdio_register(struc
  
  	if (external) {
  		mv88e6xxx_reg_lock(chip);
diff --git a/target/linux/generic/backport-6.6/901-v6.13-net-dsa-mv88e6xxx-Support-LED-control.patch b/target/linux/generic/backport-6.6/901-v6.13-net-dsa-mv88e6xxx-Support-LED-control.patch
index 1674359d266634..838cce8f4a03ff 100644
--- a/target/linux/generic/backport-6.6/901-v6.13-net-dsa-mv88e6xxx-Support-LED-control.patch
+++ b/target/linux/generic/backport-6.6/901-v6.13-net-dsa-mv88e6xxx-Support-LED-control.patch
@@ -127,7 +127,7 @@ Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
  #include <linux/netdevice.h>
  #include <linux/gpio/consumer.h>
  #include <linux/phylink.h>
-@@ -3265,14 +3266,43 @@ static int mv88e6xxx_setup_upstream_port
+@@ -3276,14 +3277,43 @@ static int mv88e6xxx_setup_upstream_port
  static int mv88e6xxx_setup_port(struct mv88e6xxx_chip *chip, int port)
  {
  	struct device_node *phy_handle = NULL;
@@ -173,7 +173,7 @@ Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
  
  	err = mv88e6xxx_port_setup_mac(chip, port, LINK_UNFORCED,
  				       SPEED_UNFORCED, DUPLEX_UNFORCED,
-@@ -4491,6 +4521,7 @@ static const struct mv88e6xxx_ops mv88e6
+@@ -4517,6 +4547,7 @@ static const struct mv88e6xxx_ops mv88e6
  	.port_disable_learn_limit = mv88e6xxx_port_disable_learn_limit,
  	.port_disable_pri_override = mv88e6xxx_port_disable_pri_override,
  	.port_get_cmode = mv88e6352_port_get_cmode,
@@ -181,7 +181,7 @@ Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
  	.port_setup_message_port = mv88e6xxx_setup_message_port,
  	.stats_snapshot = mv88e6320_g1_stats_snapshot,
  	.stats_set_histogram = mv88e6095_g1_stats_set_histogram,
-@@ -4593,6 +4624,7 @@ static const struct mv88e6xxx_ops mv88e6
+@@ -4619,6 +4650,7 @@ static const struct mv88e6xxx_ops mv88e6
  	.port_disable_learn_limit = mv88e6xxx_port_disable_learn_limit,
  	.port_disable_pri_override = mv88e6xxx_port_disable_pri_override,
  	.port_get_cmode = mv88e6352_port_get_cmode,
@@ -189,7 +189,7 @@ Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
  	.port_setup_message_port = mv88e6xxx_setup_message_port,
  	.stats_snapshot = mv88e6320_g1_stats_snapshot,
  	.stats_set_histogram = mv88e6095_g1_stats_set_histogram,
-@@ -4868,6 +4900,7 @@ static const struct mv88e6xxx_ops mv88e6
+@@ -4894,6 +4926,7 @@ static const struct mv88e6xxx_ops mv88e6
  	.port_disable_learn_limit = mv88e6xxx_port_disable_learn_limit,
  	.port_disable_pri_override = mv88e6xxx_port_disable_pri_override,
  	.port_get_cmode = mv88e6352_port_get_cmode,
@@ -197,7 +197,7 @@ Signed-off-by: Linus Walleij <linus.walleij@linaro.org>
  	.port_setup_message_port = mv88e6xxx_setup_message_port,
  	.stats_snapshot = mv88e6320_g1_stats_snapshot,
  	.stats_set_histogram = mv88e6095_g1_stats_set_histogram,
-@@ -5290,6 +5323,7 @@ static const struct mv88e6xxx_ops mv88e6
+@@ -5318,6 +5351,7 @@ static const struct mv88e6xxx_ops mv88e6
  	.port_disable_learn_limit = mv88e6xxx_port_disable_learn_limit,
  	.port_disable_pri_override = mv88e6xxx_port_disable_pri_override,
  	.port_get_cmode = mv88e6352_port_get_cmode,
diff --git a/target/linux/generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch b/target/linux/generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch
index c1f595913100df..85cd79f1a9ddd1 100644
--- a/target/linux/generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch
+++ b/target/linux/generic/hack-6.6/610-net-page_pool-try-to-free-deferred-skbs-while-waitin.patch
@@ -16,7 +16,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 
 --- a/net/core/page_pool.c
 +++ b/net/core/page_pool.c
-@@ -873,12 +873,23 @@ static void page_pool_release_retry(stru
+@@ -873,7 +873,7 @@ static void page_pool_release_retry(stru
  {
  	struct delayed_work *dwq = to_delayed_work(wq);
  	struct page_pool *pool = container_of(dwq, typeof(*pool), release_dw);
@@ -24,7 +24,9 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +	int cpu, inflight;
  
  	inflight = page_pool_release(pool);
- 	if (!inflight)
+ 	/* In rare cases, a driver bug may cause inflight to go negative.
+@@ -885,6 +885,17 @@ static void page_pool_release_retry(stru
+ 	if (inflight <= 0)
  		return;
  
 +	/* Run NET_RX_SOFTIRQ in order to free pending skbs in softnet_data
diff --git a/target/linux/generic/hack-6.6/711-net-dsa-mv88e6xxx-disable-ATU-violation.patch b/target/linux/generic/hack-6.6/711-net-dsa-mv88e6xxx-disable-ATU-violation.patch
index ab594a471e3c13..7c62435d6ee7e9 100644
--- a/target/linux/generic/hack-6.6/711-net-dsa-mv88e6xxx-disable-ATU-violation.patch
+++ b/target/linux/generic/hack-6.6/711-net-dsa-mv88e6xxx-disable-ATU-violation.patch
@@ -9,7 +9,7 @@ Subject: [PATCH] net/dsa/mv88e6xxx: disable ATU violation
 
 --- a/drivers/net/dsa/mv88e6xxx/chip.c
 +++ b/drivers/net/dsa/mv88e6xxx/chip.c
-@@ -3435,6 +3435,9 @@ static int mv88e6xxx_setup_port(struct m
+@@ -3446,6 +3446,9 @@ static int mv88e6xxx_setup_port(struct m
  	else
  		reg = 1 << port;
  
diff --git a/target/linux/generic/hack-6.6/760-net-usb-r8152-add-LED-configuration-from-OF.patch b/target/linux/generic/hack-6.6/760-net-usb-r8152-add-LED-configuration-from-OF.patch
index 069c6823033594..680440b310c037 100644
--- a/target/linux/generic/hack-6.6/760-net-usb-r8152-add-LED-configuration-from-OF.patch
+++ b/target/linux/generic/hack-6.6/760-net-usb-r8152-add-LED-configuration-from-OF.patch
@@ -22,7 +22,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  #include <linux/crc32.h>
  #include <linux/if_vlan.h>
  #include <linux/uaccess.h>
-@@ -7044,6 +7045,22 @@ static void rtl_tally_reset(struct r8152
+@@ -7045,6 +7046,22 @@ static void rtl_tally_reset(struct r8152
  	ocp_write_word(tp, MCU_TYPE_PLA, PLA_RSTTALLY, ocp_data);
  }
  
@@ -45,7 +45,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  static void r8152b_init(struct r8152 *tp)
  {
  	u32 ocp_data;
-@@ -7085,6 +7102,8 @@ static void r8152b_init(struct r8152 *tp
+@@ -7086,6 +7103,8 @@ static void r8152b_init(struct r8152 *tp
  	ocp_data = ocp_read_word(tp, MCU_TYPE_USB, USB_USB_CTRL);
  	ocp_data &= ~(RX_AGG_DISABLE | RX_ZERO_EN);
  	ocp_write_word(tp, MCU_TYPE_USB, USB_USB_CTRL, ocp_data);
@@ -54,7 +54,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  }
  
  static void r8153_init(struct r8152 *tp)
-@@ -7225,6 +7244,8 @@ static void r8153_init(struct r8152 *tp)
+@@ -7226,6 +7245,8 @@ static void r8153_init(struct r8152 *tp)
  		tp->coalesce = COALESCE_SLOW;
  		break;
  	}
@@ -63,7 +63,7 @@ Signed-off-by: David Bauer <mail@david-bauer.net>
  }
  
  static void r8153b_init(struct r8152 *tp)
-@@ -7307,6 +7328,8 @@ static void r8153b_init(struct r8152 *tp
+@@ -7308,6 +7329,8 @@ static void r8153b_init(struct r8152 *tp
  	rtl_tally_reset(tp);
  
  	tp->coalesce = 15000;	/* 15 us */
diff --git a/target/linux/generic/kernel-6.6 b/target/linux/generic/kernel-6.6
index e3de08df98dee6..07ba30cb3aaf51 100644
--- a/target/linux/generic/kernel-6.6
+++ b/target/linux/generic/kernel-6.6
@@ -1,2 +1,2 @@
-LINUX_VERSION-6.6 = .87
-LINUX_KERNEL_HASH-6.6.87 = 8957e5c2dacdbc47a16dbf1f6303ca7088409be6197a3881f752313275357ac6
+LINUX_VERSION-6.6 = .88
+LINUX_KERNEL_HASH-6.6.88 = 19df89b63ef7e950de7297dabfac0569183bf87636f4c300a25336c7da490650
diff --git a/target/linux/generic/pending-6.6/510-block-add-uImage.FIT-subimage-block-driver.patch b/target/linux/generic/pending-6.6/510-block-add-uImage.FIT-subimage-block-driver.patch
index 056b49db3ef9c0..1ad221035285ba 100644
--- a/target/linux/generic/pending-6.6/510-block-add-uImage.FIT-subimage-block-driver.patch
+++ b/target/linux/generic/pending-6.6/510-block-add-uImage.FIT-subimage-block-driver.patch
@@ -36,7 +36,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -22016,6 +22016,12 @@ F:	Documentation/filesystems/ubifs-authe
+@@ -22017,6 +22017,12 @@ F:	Documentation/filesystems/ubifs-authe
  F:	Documentation/filesystems/ubifs.rst
  F:	fs/ubifs/
  
diff --git a/target/linux/generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch b/target/linux/generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch
index 4b4825ae3b9a4a..2f4418d4f2efe7 100644
--- a/target/linux/generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch
+++ b/target/linux/generic/pending-6.6/620-net_sched-codel-do-not-defer-queue-length-update.patch
@@ -22,19 +22,16 @@ Link: https://bugzilla.kernel.org/show_bug.cgi?id=109581
 
 --- a/net/sched/sch_codel.c
 +++ b/net/sched/sch_codel.c
-@@ -95,11 +95,17 @@ static struct sk_buff *codel_qdisc_deque
+@@ -95,8 +95,17 @@ static struct sk_buff *codel_qdisc_deque
  			    &q->stats, qdisc_pkt_len, codel_get_enqueue_time,
  			    drop_func, dequeue_func);
  
--	/* We cant call qdisc_tree_reduce_backlog() if our qlen is 0,
--	 * or HTB crashes. Defer it for next round.
 +	/* If our qlen is 0 qdisc_tree_reduce_backlog() will deactivate
 +	 * parent class, dequeue in parent qdisc will do the same if we
 +	 * return skb. Temporary increment qlen if we have skb.
- 	 */
--	if (q->stats.drop_count && sch->q.qlen) {
++	 */
+ 	if (q->stats.drop_count) {
 -		qdisc_tree_reduce_backlog(sch, q->stats.drop_count, q->stats.drop_len);
-+	if (q->stats.drop_count) {
 +		if (skb)
 +			sch->q.qlen++;
 +		qdisc_tree_reduce_backlog(sch, q->stats.drop_count,
@@ -68,14 +65,12 @@ Link: https://bugzilla.kernel.org/show_bug.cgi?id=109581
  	if (!skb) {
  		/* force a pass through old_flows to prevent starvation */
  		if ((head == &q->new_flows) && !list_empty(&q->old_flows))
-@@ -314,15 +329,6 @@ begin:
+@@ -314,13 +329,6 @@ begin:
  	}
  	qdisc_bstats_update(sch, skb);
  	flow->deficit -= qdisc_pkt_len(skb);
--	/* We cant call qdisc_tree_reduce_backlog() if our qlen is 0,
--	 * or HTB crashes. Defer it for next round.
--	 */
--	if (q->cstats.drop_count && sch->q.qlen) {
+-
+-	if (q->cstats.drop_count) {
 -		qdisc_tree_reduce_backlog(sch, q->cstats.drop_count,
 -					  q->cstats.drop_len);
 -		q->cstats.drop_count = 0;
diff --git a/target/linux/generic/pending-6.6/681-net-remove-NETIF_F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch b/target/linux/generic/pending-6.6/681-net-remove-NETIF_F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch
index 58ccff247fb5c5..ff0a9ca5ae0c8d 100644
--- a/target/linux/generic/pending-6.6/681-net-remove-NETIF_F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch
+++ b/target/linux/generic/pending-6.6/681-net-remove-NETIF_F_GSO_FRAGLIST-from-NETIF_F_GSO_SOF.patch
@@ -76,7 +76,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  	if ((ret & NETIF_F_GSO_ENCAP_ALL) && (ret & NETIF_F_CSUM_MASK))
 --- a/net/8021q/vlan_dev.c
 +++ b/net/8021q/vlan_dev.c
-@@ -583,7 +583,7 @@ static int vlan_dev_init(struct net_devi
+@@ -560,7 +560,7 @@ static int vlan_dev_init(struct net_devi
  		dev->state |= (1 << __LINK_STATE_NOCARRIER);
  
  	dev->hw_features = NETIF_F_HW_CSUM | NETIF_F_SG |
@@ -85,7 +85,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
  			   NETIF_F_GSO_ENCAP_ALL |
  			   NETIF_F_HIGHDMA | NETIF_F_SCTP_CRC |
  			   NETIF_F_ALL_FCOE;
-@@ -676,7 +676,7 @@ static netdev_features_t vlan_dev_fix_fe
+@@ -653,7 +653,7 @@ static netdev_features_t vlan_dev_fix_fe
  	if (lower_features & (NETIF_F_IP_CSUM|NETIF_F_IPV6_CSUM))
  		lower_features |= NETIF_F_HW_CSUM;
  	features = netdev_intersect_features(features, lower_features);
diff --git a/target/linux/generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch b/target/linux/generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch
index ccb9d1bc1d06eb..b2dd3c85f65381 100644
--- a/target/linux/generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch
+++ b/target/linux/generic/pending-6.6/734-net-ethernet-mediatek-enlarge-DMA-reserve-buffer.patch
@@ -34,7 +34,7 @@ Signed-off-by: Chad Monroe <chad@monroe.io>
 +			val &= ~MTK_RESV_BUF_MASK;
  			val |= MTK_MUTLI_CNT | MTK_RESV_BUF |
  			       MTK_WCOMP_EN | MTK_DMAD_WR_WDONE |
- 			       MTK_CHK_DDONE_EN | MTK_LEAKY_BUCKET_EN;
+ 			       MTK_CHK_DDONE_EN;
 -		else
 +		} else {
  			val |= MTK_RX_BT_32DWORDS;
diff --git a/target/linux/generic/pending-6.6/739-05-net-pcs-add-driver-for-MediaTek-USXGMII-PCS.patch b/target/linux/generic/pending-6.6/739-05-net-pcs-add-driver-for-MediaTek-USXGMII-PCS.patch
index 0aaa39717c110d..f3716c6c2d505c 100644
--- a/target/linux/generic/pending-6.6/739-05-net-pcs-add-driver-for-MediaTek-USXGMII-PCS.patch
+++ b/target/linux/generic/pending-6.6/739-05-net-pcs-add-driver-for-MediaTek-USXGMII-PCS.patch
@@ -19,7 +19,7 @@ Signed-off-by: Daniel Golle <daniel@makrotopia.org>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -13356,7 +13356,9 @@ M:	Daniel Golle <daniel@makrotopia.org>
+@@ -13357,7 +13357,9 @@ M:	Daniel Golle <daniel@makrotopia.org>
  L:	netdev@vger.kernel.org
  S:	Maintained
  F:	drivers/net/pcs/pcs-mtk-lynxi.c
diff --git a/target/linux/ipq40xx/patches-6.6/700-net-ipqess-introduce-the-Qualcomm-IPQESS-driver.patch b/target/linux/ipq40xx/patches-6.6/700-net-ipqess-introduce-the-Qualcomm-IPQESS-driver.patch
index b8557bacc75865..39e1bceefc3fae 100644
--- a/target/linux/ipq40xx/patches-6.6/700-net-ipqess-introduce-the-Qualcomm-IPQESS-driver.patch
+++ b/target/linux/ipq40xx/patches-6.6/700-net-ipqess-introduce-the-Qualcomm-IPQESS-driver.patch
@@ -32,7 +32,7 @@ Signed-off-by: Maxime Chevallier <maxime.chevallier@bootlin.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -17717,6 +17717,13 @@ L:	netdev@vger.kernel.org
+@@ -17718,6 +17718,13 @@ L:	netdev@vger.kernel.org
  S:	Maintained
  F:	drivers/net/ethernet/qualcomm/emac/
  
diff --git a/target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch b/target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch
index 469f669bd47915..8c7567fa9382a9 100644
--- a/target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch
+++ b/target/linux/ipq40xx/patches-6.6/701-net-dsa-add-out-of-band-tagging-protocol.patch
@@ -64,7 +64,7 @@ Signed-off-by: Maxime Chevallier <maxime.chevallier@bootlin.com>
  
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -17723,6 +17723,7 @@ L:	netdev@vger.kernel.org
+@@ -17724,6 +17724,7 @@ L:	netdev@vger.kernel.org
  S:	Maintained
  F:	Documentation/devicetree/bindings/net/qcom,ipq4019-ess-edma.yaml
  F:	drivers/net/ethernet/qualcomm/ipqess/
diff --git a/target/linux/lantiq/patches-6.6/0731-v6.11-dt-bindings-net-dsa-lantiq-gswip-convert-to-YAML-schema.patch b/target/linux/lantiq/patches-6.6/0731-v6.11-dt-bindings-net-dsa-lantiq-gswip-convert-to-YAML-schema.patch
index 40e52f2812bb36..34a730c36fead3 100644
--- a/target/linux/lantiq/patches-6.6/0731-v6.11-dt-bindings-net-dsa-lantiq-gswip-convert-to-YAML-schema.patch
+++ b/target/linux/lantiq/patches-6.6/0731-v6.11-dt-bindings-net-dsa-lantiq-gswip-convert-to-YAML-schema.patch
@@ -382,7 +382,7 @@ Signed-off-by: Jakub Kicinski <kuba@kernel.org>
 -};
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -11863,6 +11863,7 @@ LANTIQ / INTEL Ethernet drivers
+@@ -11864,6 +11864,7 @@ LANTIQ / INTEL Ethernet drivers
  M:	Hauke Mehrtens <hauke@hauke-m.de>
  L:	netdev@vger.kernel.org
  S:	Maintained
diff --git a/target/linux/layerscape/patches-6.6/701-staging-add-fsl_ppfe-driver.patch b/target/linux/layerscape/patches-6.6/701-staging-add-fsl_ppfe-driver.patch
index 764d29f10d9e17..59bbb964c5187f 100644
--- a/target/linux/layerscape/patches-6.6/701-staging-add-fsl_ppfe-driver.patch
+++ b/target/linux/layerscape/patches-6.6/701-staging-add-fsl_ppfe-driver.patch
@@ -891,7 +891,7 @@ Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
 +};
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -8359,6 +8359,14 @@ F:	drivers/ptp/ptp_qoriq.c
+@@ -8360,6 +8360,14 @@ F:	drivers/ptp/ptp_qoriq.c
  F:	drivers/ptp/ptp_qoriq_debugfs.c
  F:	include/linux/fsl/ptp_qoriq.h
  
diff --git a/target/linux/mediatek/patches-6.6/806-v6.9-pwm-mediatek-add-support-for-MT7988.patch b/target/linux/mediatek/patches-6.6/806-v6.9-pwm-mediatek-add-support-for-MT7988.patch
index 01768ff01fdb9a..946f3a525280ab 100644
--- a/target/linux/mediatek/patches-6.6/806-v6.9-pwm-mediatek-add-support-for-MT7988.patch
+++ b/target/linux/mediatek/patches-6.6/806-v6.9-pwm-mediatek-add-support-for-MT7988.patch
@@ -20,7 +20,7 @@ Signed-off-by: Uwe Kleine-Knig <u.kleine-koenig@pengutronix.de>
 
 --- a/drivers/pwm/pwm-mediatek.c
 +++ b/drivers/pwm/pwm-mediatek.c
-@@ -341,6 +341,13 @@ static const struct pwm_mediatek_of_data
+@@ -345,6 +345,13 @@ static const struct pwm_mediatek_of_data
  	.reg_offset = mtk_pwm_reg_offset_v1,
  };
  
@@ -34,7 +34,7 @@ Signed-off-by: Uwe Kleine-Knig <u.kleine-koenig@pengutronix.de>
  static const struct pwm_mediatek_of_data mt8183_pwm_data = {
  	.num_pwms = 4,
  	.pwm45_fixup = false,
-@@ -371,6 +378,7 @@ static const struct of_device_id pwm_med
+@@ -375,6 +382,7 @@ static const struct of_device_id pwm_med
  	{ .compatible = "mediatek,mt7629-pwm", .data = &mt7629_pwm_data },
  	{ .compatible = "mediatek,mt7981-pwm", .data = &mt7981_pwm_data },
  	{ .compatible = "mediatek,mt7986-pwm", .data = &mt7986_pwm_data },
diff --git a/target/linux/mvebu/patches-6.6/907-MAINTAINERS-Add-an-entry-for-the-IEI-WT61P803-PUZZLE.patch b/target/linux/mvebu/patches-6.6/907-MAINTAINERS-Add-an-entry-for-the-IEI-WT61P803-PUZZLE.patch
index dad8c16be1d340..d70bfe9a1af0ec 100644
--- a/target/linux/mvebu/patches-6.6/907-MAINTAINERS-Add-an-entry-for-the-IEI-WT61P803-PUZZLE.patch
+++ b/target/linux/mvebu/patches-6.6/907-MAINTAINERS-Add-an-entry-for-the-IEI-WT61P803-PUZZLE.patch
@@ -16,7 +16,7 @@ Cc: Robert Marko <robert.marko@sartura.hr>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -10142,6 +10142,22 @@ IFCVF VIRTIO DATA PATH ACCELERATOR
+@@ -10143,6 +10143,22 @@ IFCVF VIRTIO DATA PATH ACCELERATOR
  R:	Zhu Lingshan <lingshan.zhu@intel.com>
  F:	drivers/vdpa/ifcvf/
  
diff --git a/target/linux/ramips/patches-6.6/810-uvc-add-iPassion-iP2970-support.patch b/target/linux/ramips/patches-6.6/810-uvc-add-iPassion-iP2970-support.patch
index 916f08ac5d73c7..01ecb16aeb196a 100644
--- a/target/linux/ramips/patches-6.6/810-uvc-add-iPassion-iP2970-support.patch
+++ b/target/linux/ramips/patches-6.6/810-uvc-add-iPassion-iP2970-support.patch
@@ -13,7 +13,7 @@ Signed-off-by: John Crispin <blogic@openwrt.org>
 
 --- a/drivers/media/usb/uvc/uvc_driver.c
 +++ b/drivers/media/usb/uvc/uvc_driver.c
-@@ -3287,6 +3287,18 @@ static const struct usb_device_id uvc_id
+@@ -3296,6 +3296,18 @@ static const struct usb_device_id uvc_id
  	  .bInterfaceSubClass	= 1,
  	  .bInterfaceProtocol	= 0,
  	  .driver_info		= UVC_INFO_META(V4L2_META_FMT_D4XX) },
diff --git a/target/linux/realtek/patches-6.6/710-net-phy-sfp-re-probe-modules-on-DEV_UP-event.patch b/target/linux/realtek/patches-6.6/710-net-phy-sfp-re-probe-modules-on-DEV_UP-event.patch
index cab2d6bf3ced84..9f9051045e34eb 100644
--- a/target/linux/realtek/patches-6.6/710-net-phy-sfp-re-probe-modules-on-DEV_UP-event.patch
+++ b/target/linux/realtek/patches-6.6/710-net-phy-sfp-re-probe-modules-on-DEV_UP-event.patch
@@ -10,7 +10,7 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
 
 --- a/drivers/net/phy/sfp.c
 +++ b/drivers/net/phy/sfp.c
-@@ -1568,6 +1568,10 @@ static void sfp_hwmon_probe(struct work_
+@@ -1570,6 +1570,10 @@ static void sfp_hwmon_probe(struct work_
  	struct sfp *sfp = container_of(work, struct sfp, hwmon_probe.work);
  	int err;
  
@@ -21,7 +21,7 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
  	/* hwmon interface needs to access 16bit registers in atomic way to
  	 * guarantee coherency of the diagnostic monitoring data. If it is not
  	 * possible to guarantee coherency because EEPROM is broken in such way
-@@ -2430,6 +2434,13 @@ static void sfp_sm_module(struct sfp *sf
+@@ -2432,6 +2436,13 @@ static void sfp_sm_module(struct sfp *sf
  		return;
  	}
  
diff --git a/target/linux/realtek/patches-6.6/714-net-phy-sfp-add-support-for-SMBus.patch b/target/linux/realtek/patches-6.6/714-net-phy-sfp-add-support-for-SMBus.patch
index 3b7a3b9db11d25..ea6149281d1cbf 100644
--- a/target/linux/realtek/patches-6.6/714-net-phy-sfp-add-support-for-SMBus.patch
+++ b/target/linux/realtek/patches-6.6/714-net-phy-sfp-add-support-for-SMBus.patch
@@ -10,7 +10,7 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
 
 --- a/drivers/net/phy/sfp.c
 +++ b/drivers/net/phy/sfp.c
-@@ -673,10 +673,64 @@ static int sfp_i2c_write(struct sfp *sfp
+@@ -675,10 +675,64 @@ static int sfp_i2c_write(struct sfp *sfp
  	return ret == ARRAY_SIZE(msgs) ? len : 0;
  }
  
@@ -77,7 +77,7 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
  
  	sfp->i2c = i2c;
  	sfp->read = sfp_i2c_read;
-@@ -708,6 +762,29 @@ static int sfp_i2c_mdiobus_create(struct
+@@ -710,6 +764,29 @@ static int sfp_i2c_mdiobus_create(struct
  	return 0;
  }
  
@@ -107,7 +107,7 @@ Signed-off-by: Antoine Tenart <antoine.tenart@bootlin.com>
  static void sfp_i2c_mdiobus_destroy(struct sfp *sfp)
  {
  	mdiobus_unregister(sfp->i2c_mii);
-@@ -1886,9 +1963,15 @@ static void sfp_sm_fault(struct sfp *sfp
+@@ -1888,9 +1965,15 @@ static void sfp_sm_fault(struct sfp *sfp
  
  static int sfp_sm_add_mdio_bus(struct sfp *sfp)
  {
diff --git a/target/linux/starfive/patches-6.6/0049-MAINTAINERS-Update-all-StarFive-entries.patch b/target/linux/starfive/patches-6.6/0049-MAINTAINERS-Update-all-StarFive-entries.patch
index dfef89a5a2e82a..5f2c94914f73be 100644
--- a/target/linux/starfive/patches-6.6/0049-MAINTAINERS-Update-all-StarFive-entries.patch
+++ b/target/linux/starfive/patches-6.6/0049-MAINTAINERS-Update-all-StarFive-entries.patch
@@ -12,7 +12,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -7053,6 +7053,14 @@ T:	git git://anongit.freedesktop.org/drm
+@@ -7054,6 +7054,14 @@ T:	git git://anongit.freedesktop.org/drm
  F:	Documentation/devicetree/bindings/display/rockchip/
  F:	drivers/gpu/drm/rockchip/
  
@@ -27,7 +27,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  DRM DRIVERS FOR STI
  M:	Alain Volmat <alain.volmat@foss.st.com>
  L:	dri-devel@lists.freedesktop.org
-@@ -16017,6 +16025,13 @@ F:	Documentation/i2c/busses/i2c-ocores.r
+@@ -16018,6 +16026,13 @@ F:	Documentation/i2c/busses/i2c-ocores.r
  F:	drivers/i2c/busses/i2c-ocores.c
  F:	include/linux/platform_data/i2c-ocores.h
  
@@ -41,7 +41,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  OPENRISC ARCHITECTURE
  M:	Jonas Bonn <jonas@southpole.se>
  M:	Stefan Kristiansson <stefan.kristiansson@saunalahti.fi>
-@@ -16428,6 +16443,14 @@ S:	Maintained
+@@ -16429,6 +16444,14 @@ S:	Maintained
  F:	Documentation/devicetree/bindings/pci/layerscape-pcie-gen4.txt
  F:	drivers/pci/controller/mobiveil/pcie-layerscape-gen4.c
  
@@ -56,7 +56,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  PCI DRIVER FOR RENESAS R-CAR
  M:	Marek Vasut <marek.vasut+renesas@gmail.com>
  M:	Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>
-@@ -16659,7 +16682,7 @@ M:	Daire McNamara <daire.mcnamara@microc
+@@ -16660,7 +16683,7 @@ M:	Daire McNamara <daire.mcnamara@microc
  L:	linux-pci@vger.kernel.org
  S:	Supported
  F:	Documentation/devicetree/bindings/pci/microchip*
@@ -65,7 +65,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  
  PCIE DRIVER FOR QUALCOMM MSM
  M:	Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
-@@ -16683,6 +16706,13 @@ S:	Maintained
+@@ -16684,6 +16707,13 @@ S:	Maintained
  F:	Documentation/devicetree/bindings/pci/socionext,uniphier-pcie*
  F:	drivers/pci/controller/dwc/pcie-uniphier*
  
@@ -79,7 +79,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  PCIE DRIVER FOR ST SPEAR13XX
  M:	Pratyush Anand <pratyush.anand@gmail.com>
  L:	linux-pci@vger.kernel.org
-@@ -18455,7 +18485,7 @@ F:	drivers/char/hw_random/mpfs-rng.c
+@@ -18456,7 +18486,7 @@ F:	drivers/char/hw_random/mpfs-rng.c
  F:	drivers/clk/microchip/clk-mpfs*.c
  F:	drivers/i2c/busses/i2c-microchip-corei2c.c
  F:	drivers/mailbox/mailbox-mpfs.c
@@ -88,7 +88,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  F:	drivers/pwm/pwm-microchip-core.c
  F:	drivers/reset/reset-mpfs.c
  F:	drivers/rtc/rtc-mpfs.c
-@@ -20436,6 +20466,15 @@ M:	Ion Badulescu <ionut@badula.org>
+@@ -20437,6 +20467,15 @@ M:	Ion Badulescu <ionut@badula.org>
  S:	Odd Fixes
  F:	drivers/net/ethernet/adaptec/starfire*
  
@@ -104,7 +104,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  STARFIVE CRYPTO DRIVER
  M:	Jia Jie Ho <jiajie.ho@starfivetech.com>
  M:	William Qiu <william.qiu@starfivetech.com>
-@@ -20474,6 +20513,13 @@ S:	Supported
+@@ -20475,6 +20514,13 @@ S:	Supported
  F:	Documentation/devicetree/bindings/clock/starfive,jh7110-pll.yaml
  F:	drivers/clk/starfive/clk-starfive-jh7110-pll.c
  
@@ -118,7 +118,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  STARFIVE JH7110 SYSCON
  M:	William Qiu <william.qiu@starfivetech.com>
  M:	Xingyu Wu <xingyu.wu@starfivetech.com>
-@@ -20521,9 +20567,10 @@ F:	drivers/usb/cdns3/cdns3-starfive.c
+@@ -20522,9 +20568,10 @@ F:	drivers/usb/cdns3/cdns3-starfive.c
  
  STARFIVE JH71XX PMU CONTROLLER DRIVER
  M:	Walker Chen <walker.chen@starfivetech.com>
@@ -130,7 +130,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  F:	include/dt-bindings/power/starfive,jh7110-pmu.h
  
  STARFIVE SOC DRIVERS
-@@ -20531,7 +20578,13 @@ M:	Conor Dooley <conor@kernel.org>
+@@ -20532,7 +20579,13 @@ M:	Conor Dooley <conor@kernel.org>
  S:	Maintained
  T:	git https://git.kernel.org/pub/scm/linux/kernel/git/conor/linux.git/
  F:	Documentation/devicetree/bindings/soc/starfive/
diff --git a/target/linux/starfive/patches-6.6/0052-ASoC-dwc-i2s-Fix-getting-platform-data-error-for-Sta.patch b/target/linux/starfive/patches-6.6/0052-ASoC-dwc-i2s-Fix-getting-platform-data-error-for-Sta.patch
index e274e84a259ddf..65ca992201db45 100644
--- a/target/linux/starfive/patches-6.6/0052-ASoC-dwc-i2s-Fix-getting-platform-data-error-for-Sta.patch
+++ b/target/linux/starfive/patches-6.6/0052-ASoC-dwc-i2s-Fix-getting-platform-data-error-for-Sta.patch
@@ -16,7 +16,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -917,7 +917,11 @@ static int jh7110_i2stx0_clk_cfg(struct
+@@ -916,7 +916,11 @@ static int jh7110_i2stx0_clk_cfg(struct
  
  static int dw_i2s_probe(struct platform_device *pdev)
  {
diff --git a/target/linux/starfive/patches-6.6/0053-ASoC-dwc-i2s-Add-RX-master-support-for-StarFive-JH71.patch b/target/linux/starfive/patches-6.6/0053-ASoC-dwc-i2s-Add-RX-master-support-for-StarFive-JH71.patch
index 8203beca52ed91..48d47dce76000d 100644
--- a/target/linux/starfive/patches-6.6/0053-ASoC-dwc-i2s-Add-RX-master-support-for-StarFive-JH71.patch
+++ b/target/linux/starfive/patches-6.6/0053-ASoC-dwc-i2s-Add-RX-master-support-for-StarFive-JH71.patch
@@ -15,7 +15,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
 +++ b/sound/soc/dwc/dwc-i2s.c
-@@ -906,6 +906,27 @@ static int jh7110_i2srx_crg_init(struct
+@@ -905,6 +905,27 @@ static int jh7110_i2srx_crg_init(struct
  	return jh7110_i2s_crg_slave_init(dev);
  }
  
@@ -43,7 +43,7 @@ Signed-off-by: Hal Feng <hal.feng@starfivetech.com>
  static int jh7110_i2stx0_clk_cfg(struct i2s_clk_config_data *config)
  {
  	struct dw_i2s_dev *dev = container_of(config, struct dw_i2s_dev, config);
-@@ -1086,11 +1107,21 @@ static const struct i2s_platform_data jh
+@@ -1085,11 +1106,21 @@ static const struct i2s_platform_data jh
  	.i2s_pd_init = jh7110_i2srx_crg_init,
  };
  
diff --git a/target/linux/starfive/patches-6.6/1014-reset-starfive-Add-JH7100-audio-reset-driver.patch b/target/linux/starfive/patches-6.6/1014-reset-starfive-Add-JH7100-audio-reset-driver.patch
index 79e7d4ddb48cbb..8b24eb33e5182e 100644
--- a/target/linux/starfive/patches-6.6/1014-reset-starfive-Add-JH7100-audio-reset-driver.patch
+++ b/target/linux/starfive/patches-6.6/1014-reset-starfive-Add-JH7100-audio-reset-driver.patch
@@ -20,7 +20,7 @@ Signed-off-by: Emil Renner Berthing <kernel@esmil.dk>
 
 --- a/MAINTAINERS
 +++ b/MAINTAINERS
-@@ -20555,7 +20555,7 @@ STARFIVE JH71X0 RESET CONTROLLER DRIVERS
+@@ -20556,7 +20556,7 @@ STARFIVE JH71X0 RESET CONTROLLER DRIVERS
  M:	Emil Renner Berthing <kernel@esmil.dk>
  M:	Hal Feng <hal.feng@starfivetech.com>
  S:	Maintained
